src/components/shapes/draw_newshape/newshapes.js

```diff
@@ -70,8 +62,65 @@ module.exports = function newShapes(outlines, dragOptions) {
         }
     }
 
-    var isOpenMode = openMode(dragmode);
+    var newShape = createShapeObj(outlines, dragOptions);
```

Comment:
The only actual change here is moving some of the logic into a separate `createShapeObj()` function